automated install/basic-install.sh

```diff
@@ -182,47 +182,43 @@ os_check() {
         # This function gets a list of supported OS versions from a TXT record at versions.pi-hole.net
         # and determines whether or not the script is running on one of those systems
-        local remote_os_domain valid_os valid_version detected_os_pretty detected_os detected_version display_warning
+        local remote_os_domain valid_os valid_version detected_os detected_version display_warning
         remote_os_domain="versions.pi-hole.net"
-        valid_os=false
-        valid_version=false
-        display_warning=true
 
-        detected_os_pretty=$(cat /etc/*release | grep PRETTY_NAME | cut -d '=' -f2- | tr -d '"')
-        detected_os="${detected_os_pretty%% *}"
-        detected_version=$(cat /etc/*release | grep VERSION_ID | cut -d '=' -f2- | tr -d '"')
+        detected_os=$(grep "\bID\b" /etc/os-release | cut -d '=' -f2 | tr -d '"')
+        detected_version=$(grep VERSION_ID /etc/os-release | cut -d '=' -f2 | tr -d '"')
 
         IFS=" " read -r -a supportedOS < <(dig +short -t txt ${remote_os_domain} @ns1.pi-hole.net | tr -d '"')
 
         if [ ${#supportedOS[@]} -eq 0 ]; then
             printf "  %b %bRetrieval of supported OS failed. Please contact support. %b\\n" "${CROSS}" "${COL_LIGHT_RED}" "${COL_NC}"
             exit 1
         else
-          for i in "${supportedOS[@]}"
-          do
-              os_part=$(echo "$i" | cut -d '=' -f1)
-              versions_part=$(echo "$i" | cut -d '=' -f2-)
-
-              if [[ "${detected_os}" =~ ${os_part} ]]; then
-                valid_os=true
-                IFS="," read -r -a supportedVer <<<"${versions_part}"
-                for x in "${supportedVer[@]}"
-                do
-                  if [[ "${detected_version}" =~ $x ]];then
-                    valid_version=true
+            for distro_and_versions in "${supportedOS[@]}"
+            do
+                distro_part="${distro_and_versions%%=*}"
+                versions_part="${distro_and_versions##*=}"
+
+                if [[ "${detected_os^^}" =~ ${distro_part^^} ]]; then
+                    valid_os=true
+                    IFS="," read -r -a supportedVer <<<"${versions_part}"
+                    for version in "${supportedVer[@]}"
+                    do
+                        if [[ "${detected_version}" =~ $version ]]; then
+                            valid_version=true
+                            break
+                        fi
+                    done
                     break
-                  fi
-                done
-                break
-              fi
+                fi
             done
-          fi
+        fi
 
         if [ "$valid_os" = true ] && [ "$valid_version" = true ]; then
             display_warning=false
         fi
 
-        if [ "$display_warning" = true ]; then
+        if [ "$display_warning" != false ]; then
             printf "  %b %bUnsupported OS detected: %s%b\\n" "${CROSS}" "${COL_LIGHT_RED}" "${detected_os_pretty}" "${COL_NC}"
```

Comment:
You removed `detected_os_pretty`, so best would be to replace it here with `detected_os + `detected_version`

Reply:
Check latest commit and resolve conversation if it is the correct fix.

Reply:
Confirmed working

![image](https://user-images.githubusercontent.com/1998970/90319442-2925f880-df30-11ea-965b-b1165f8e045c.png)